lib/atlas/csv_document.rb

```diff
@@ -48,13 +48,53 @@ def initialize(path, normalizer = KEY_NORMALIZER)
       @table = CSV.table(@path.to_s, {
         converters: [YEAR_NORMALIZER, :all],
         header_converters: [normalizer],
+        # Needed to retrieve the headers in case
+        # of an otherwise empty csv file
+        return_headers: true
       })
 
-      @headers = @table.headers
+      @headers = table.headers
+
+      # Delete the header row for the internal representation -
+      # will be dynamically (re-)created when outputting
+      table.delete(0)
 
       raise(BlankCSVHeaderError, path) if @headers.any?(&:nil?)
     end
 
+    # Public: Creates a new csv file on disk and a corresponding
+    # new CSV document instance connected to that
+    #
+    # path    - Path to the new CSV file
+    # headers - The column headers of the new CSV
+    #
+    # Returns a CSVDocument
+    def self.create(path, headers, normalizer = KEY_NORMALIZER)
+      headers = headers.map(&normalizer)
+      initial_table = CSV::Table.new([CSV::Row.new(headers, headers, true)])
+
+      write(initial_table, Pathname.new(path))
+
```

Comment:
The current use of this method (in `TimeCurveScaler`) looks like: CSVDocument.create → add data → save.

That involves two writes to disk (in `create` and again in `save`), and if something goes wrong in "add data" or the second save, you end up with a half-complete CSV file on disk. Perhaps `create` could yield itself prior to writing, so that the user can add their initial data? (like `File.open(path, 'w') { |f| ... }`)

```ruby
CSVDocument.create(path, headers) do |doc|
  doc.set(:a, :b, 1)
  doc.set(:c, :d, 2)
end
```